extension/src/helpers/metrics.ts

```diff
@@ -0,0 +1,108 @@
+import throttle from "lodash/throttle";
+import { Middleware, AnyAction } from "redux";
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+type metricHandler<AppState> = (state: AppState, action: AnyAction) => void;
+const handlersLookup: { [key: string]: metricHandler<any>[] } = {};
+
+/*
+ * metricsMiddleware is a redux middleware that calls handlers specified to
+ * respond to a specific action type. For each action dispatched, it gets a list
+ * of registered handlers and passes the current state and action. These are
+ * intended for metrics emission, nothing else.
+ */
+export function metricsMiddleware<State>(): Middleware<{}, State> {
+  return ({ getState }) => (next) => (action: AnyAction) => {
+    const state = getState();
+    // If the `?.` fails, this ends up as `null` which appears to make eslint
+    // grumpy
+    // eslint-disable-next-line no-unused-expressions
+    handlersLookup[action.type]?.forEach((handler) => handler(state, action));
```

Comment:
Might be better to do:
```
(handlersLookup[action.type] || []).forEach((handler) => handler(state, action));
```
not sure if that will fix the error

OR if you wanna get fancy: https://github.com/tc39/proposal-nullish-coalescing
But i think nullish-coalescing might require a babel plugin for older versions of chrome <80 (and maybe Brave?)

Reply:
Since Chrome is evergreen I'm not too worried about older versions, good point tho. This seems like a corner case in the `no-unused-expressions` rule so I didn't feel bad about disabling it, but I'll swap to `|| []`

Reply:
oh, looks like you haven't added this yet. Nbd, i approved. You can add when u have a sec